lib/iris/cube.py

```diff
@@ -772,6 +772,23 @@ def __init__(self, data, standard_name=None, long_name=None,
             for cell_measure, dims in cell_measures_and_dims:
                 self.add_cell_measure(cell_measure, dims)
 
+        # When True indexing may result in a view onto the original data array,
+        # to avoid unnecessary copying.
+        self._share_data = False
+
+    @property
+    def share_data(self):
+        """Share cube data when slicing/indexing cube if True."""
+        return self._share_data
+
+    @share_data.setter
+    def share_data(self, value):
+        # Realise the data if is hasn't already been as sharing lazy data is
+        # not right now possible or a usecase understood.
+        if self.has_lazy_data():
+            _ = self.data
```

Comment:
@cpelley I really disagree that if you want to share the data, you need to realise the data **here** at this point in the property. This is bad practice IMHO.

Your effectively applying a side-effect here to suit your needs. Surely, if you do require to realise the data, then it should only occur in the code base where `share_data` is `True` and it actually needs the data at that point in time. We're now in the situation where we've thrown away any laziness, all because we've set this property .... and the kicker is that the data is loaded regardless of the `value` passed ... 

Loading the data is a BIG deal, and it's not even advertised in the doc-string. Why?

... and where's the documentation to cover this change, so that users know what's happening? No doc-string, no documentation. We really need to be always concious of our users.